Src/unstable/library/network/openssl/src/crypto/asymetric/algoritms/ssl_rsa.e

```diff
@@ -147,13 +147,58 @@ feature -- Access: Public Encrypt - Private Decrypt
 			create Result.make_from_string (l_decrypt.string)
 		end
 
-
 feature -- Sign
 
-	sign (a_priv_key: SSL_RSA_PRIVATE_KEY; a_message: READABLE_STRING_GENERAL): detachable STRING_8
+	sign_sha256 (a_priv_key: SSL_RSA_PRIVATE_KEY; a_message: READABLE_STRING_GENERAL): detachable STRING_8
+			-- Create a signed digest in a Base64 encoded string using the hash function sha256
+		do
+			Result := sign_implementation (a_priv_key, a_message, {SSL_CRYPTO_EXTERNALS}.c_evp_sha256)
+		end
+
+	sign_sha384 (a_priv_key: SSL_RSA_PRIVATE_KEY; a_message: READABLE_STRING_GENERAL): detachable STRING_8
+			-- Create a signed digest in a Base64 encoded string using the hash function sha384
+		do
+			Result := sign_implementation (a_priv_key, a_message, {SSL_CRYPTO_EXTERNALS}.c_evp_sha384)
+		end
+
+	sign_sha512 (a_priv_key: SSL_RSA_PRIVATE_KEY; a_message: READABLE_STRING_GENERAL): detachable STRING_8
+			-- Create a signed digest in a Base64 encoded string using the hash function sha512
+		do
+			Result := sign_implementation (a_priv_key, a_message, {SSL_CRYPTO_EXTERNALS}.c_evp_sha512)
+		end
+
+feature -- Verify
+
+	verify_sha256 (a_pub_key: SSL_RSA_PUBLIC_KEY; a_plain_text:READABLE_STRING_GENERAL; a_signature: READABLE_STRING_8): BOOLEAN
+			-- Ensures that the signature matches the original code.
+			-- Using the public key `a_pub_key' and the signature encoded in BASE64 with the origin text `a_plain_text'.
+			-- usign the same algorithm as the author of the signature.
+		do
+			Result := verify_implementation (a_pub_key, a_plain_text, a_signature, {SSL_CRYPTO_EXTERNALS}.c_evp_sha256)
+		end
+
+	verify_sha384 (a_pub_key: SSL_RSA_PUBLIC_KEY; a_plain_text:READABLE_STRING_GENERAL; a_signature: READABLE_STRING_8): BOOLEAN
+			-- Ensures that the signature matches the original code.
+			-- Using the public key `a_pub_key' and the signature encoded in BASE64 with the origin text `a_plain_text'.
+			-- usign the same algorithm as the author of the signature.
+		do
+			Result := verify_implementation (a_pub_key, a_plain_text, a_signature, {SSL_CRYPTO_EXTERNALS}.c_evp_sha384)
+		end
+
+	verify_sha512 (a_pub_key: SSL_RSA_PUBLIC_KEY; a_plain_text:READABLE_STRING_GENERAL; a_signature: READABLE_STRING_8): BOOLEAN
+			-- Ensures that the signature matches the original code.
+			-- Using the public key `a_pub_key' and the signature encoded in BASE64 with the origin text `a_plain_text'.
+			-- usign the same algorithm as the author of the signature.
+		do
+			Result := verify_implementation (a_pub_key, a_plain_text, a_signature, {SSL_CRYPTO_EXTERNALS}.c_evp_sha512)
+		end
+
+feature {NONE} -- Implementation
+
+	sign_implementation (a_priv_key: SSL_RSA_PRIVATE_KEY; a_message: READABLE_STRING_GENERAL; a_sha: POINTER): detachable STRING_8
```

Comment:
I would have named it  `imp_sign` or `implementation_sign`.. or just `sign`...
here it looks ... "sign the implementation.."

same comment for `verify`